org.eclipse.debug.core/core/org/eclipse/debug/internal/core/LaunchConfiguration.java

```diff
@@ -585,10 +589,27 @@ public IResource[] getMappedResources() throws CoreException {
 		return list.toArray(new IResource[list.size()]);
 	}
 
+	private static String getLineDelimiterPreference(IFile file) {
+		IScopeContext[] scopeContext;
+		if (file != null && file.getProject() != null) {
+			// project preference
+			scopeContext = new IScopeContext[] {
+					new ProjectScope(file.getProject()) };
```

Comment:
does it work if that project is closed (not accessible)?

Reply:
Good idea to test this. So, after checking, there are two things:

- When the launch configuration is saved to a common file, there is no way to edit while the project is closed.
- When the launch configuration is in the workspace, you can turn off the "filter closed projects" in the dialogue and make changes. That works (the file is updated properly).
- I do see an exception in this case when selecting the launch configuration, but that occurs both before and after my change but has not effect on the functionally to begin with. I am posting the stack trace below (it seems to happen when preparing the dialogue for rendering):

```
org.eclipse.core.runtime.CoreException: Launch configuration Main_workspace references closed project java_project
	at org.eclipse.jdt.launching.JavaRuntime.abort(JavaRuntime.java:1861)
	at org.eclipse.jdt.launching.JavaRuntime.getJavaProject(JavaRuntime.java:1733)
	at org.eclipse.jdt.launching.JavaRuntime.computeVMInstall(JavaRuntime.java:1775)
	at org.eclipse.jdt.launching.JavaRuntime.isModularConfiguration(JavaRuntime.java:1054)
	at org.eclipse.jdt.launching.StandardClasspathProvider.computeUnresolvedClasspath(StandardClasspathProvider.java:43)
	at org.eclipse.jdt.launching.JavaRuntime.computeUnresolvedRuntimeClasspath(JavaRuntime.java:1647)
	at org.eclipse.jdt.debug.ui.launchConfigurations.JavaClasspathTab.createClasspathModel(JavaClasspathTab.java:290)
	at org.eclipse.jdt.debug.ui.launchConfigurations.JavaClasspathTab.refresh(JavaClasspathTab.java:278)
	at org.eclipse.jdt.debug.ui.launchConfigurations.JavaClasspathTab.initializeFrom(JavaClasspathTab.java:241)
	at org.eclipse.debug.ui.AbstractLaunchConfigurationTabGroup.initializeFrom(AbstractLaunchConfigurationTabGroup.java:93)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationTabGroupWrapper.initializeFrom(LaunchConfigurationTabGroupWrapper.java:182)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationTabGroupViewer.displayInstanceTabs(LaunchConfigurationTabGroupViewer.java:867)
	at org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationTabGroupViewer.lambda$4(LaunchConfigurationTabGroupViewer.java:727)
```